packages/react-magma-dom/src/components/Dropdown/DropdownExpandableMenuGroup.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { Accordion, AccordionProps } from '../Accordion';
+import { DropdownContext } from './Dropdown';
+
+const StyledAccordion = styled(Accordion)<{
+  testId?: string;
+}>`
+  border: none;
+`;
+
+export interface DropdownExpandableContext {
+  hasIcon?: boolean;
+  isExpandablePanel?: boolean;
+}
+
+export const DropdownExpandableContext =
+  React.createContext<DropdownExpandableContext>({});
+
+export const DropdownExpandableMenuGroup = React.forwardRef<
+  HTMLDivElement,
+  AccordionProps
+>(props => {
+  const { children, testId, ...other } = props;
+
+  const context = React.useContext(DropdownContext);
+
+  let hasIcon = false;
+  let isExpandablePanel = false;
+
+  React.Children.forEach(children, (child: any) => {
+    if (child.type?.displayName === 'DropdownExpandableMenuItem') {
+      React.Children.forEach(child.props.children, (c: any) => {
+        if (c.type?.displayName === 'DropdownExpandableMenuButton') {
+          if (c.props.icon) {
+            hasIcon = true;
+            return;
+          }
+        }
+      });
+      if (React.isValidElement(child)) {
+        isExpandablePanel = true;
```

Comment:
Do we actually need and use `isExpandablePanel`?

Reply:
Help designate in the expandable component within the `DropdownMenuItem` for styling purposes.

Reply:
In the `DropdownMenuItem` file I mentioned a way that you may be able to check this without having to save this in the context

Reply:
This doesn't seem to work properly without context. I did try your mentioned check within `DropdownMenuItem`. 🤔